stock_split_picking/models/stock_picking.py

```diff
@@ -49,6 +49,8 @@ def split_process(self):
                             try:
                                 move_line.write(
                                     {'product_uom_qty': move_line.qty_done})
+                                product_qty_done[move_line.product_id.id] = \
+                                    move_line.qty_done
```

Comment:
I have tested in the UI and programmatically in unit tests: an error is raised when trying to create several moves with the same product id in the same picking.
As a result I think using the product id as index will not cause conflicts between moves.